packages/shared/src/transform-value.js

```diff
@@ -24,26 +24,67 @@ export default function transformValue(
       ? String(Math.round(rawValue * 10000) / 10000) + getNumberSuffix(key)
       : rawValue;
 
-  // content is one of the values that needs to wrapped in quotes.
-  // Users may write `''` without thinking about it, so we fix that.
   if (
     (key === 'content' ||
       key === 'hyphenateCharacter' ||
       key === 'hyphenate-character') &&
     typeof value === 'string'
   ) {
     const val = value.trim();
-    if (val.match(/^attr\([a-zA-Z0-9-]+\)$/)) {
-      return val;
-    }
+
+    const cssContentFunctions = [
+      'attr',
+      'counter',
+      'counters',
+      'url',
+      'linear-gradient',
+      'image-set',
+    ];
+
+    const cssContentKeywords = new Set([
+      'normal',
+      'none',
+      'open-quote',
+      'close-quote',
+      'no-open-quote',
+      'no-close-quote',
+      'inherit',
+      'initial',
+      'revert',
+      'revert-layer',
+      'unset',
+    ]);
+
+    const hasCssFunction = cssContentFunctions.some(
+      (func) => val.startsWith(`${func}(`) && val.endsWith(')'),
+    );
+
+    const isKeyword = cssContentKeywords.has(val);
+
+    const containsMixedContent = val.split(' ').some((part) => {
```

Comment:
Sadly, this won't work. I double checked to be sure and the values for `content` don't have to be separated by spaces.

There is also the concern that functions and string can contain spaces within.